### ISSUE TITLE
Spine Toolbox as an environment manager

### ISSUE THREAD

**Issue by PekkaSavolainen** [Spine-project/Spine-Toolbox#1499]
Python Tool specs now may require a kernel spec or a conda kernel spec for execution. Sharing environment(s) as part of the project for Python/Julia Tool specifications is not possible at the moment. The required environment(s) need to be saved as part of the project. When a project is loaded, there needs to be a framework for checking if user already has an environment that matches the required environment. If environment(s) do not exist, Spine Toolbox could at first instruct the user to create an environment that is compatible with a Python Tool spec. Later, Spine Toolbox could install the environments if user so chooses.

Tasks:
- [ ] Save Python environment(s) to Spine Toolbox project (requirements.txt or environment.yml for conda)
- [ ] Save Julia environment(s) (Julia project files) to Spine Toolbox project
- [ ] Build a framework for matching the requested environment to existing environments
- [ ] When loading a project, Toolbox needs to instruct the user to install requested environment(s)

**Comment by jkiviluo** [comment]
Pekka, is this done?

**Comment by PekkaSavolainen** [comment]
No but part of this will be implemented in #2554. I'm not sure if the 'automatic environment matching' framework is feasible. Maybe a bit too ambitious.

**Comment by ptsavol** [comment]
Check Inline Script Metadata https://peps.python.org/pep-0723/

Also this https://github.com/spine-tools/spine-conductor/blob/master/orchestra/runner.py (see mkvenv and other related functions in this module)

**Comment by jkiviluo** [comment]
Ignore those commit/pull request references. I copy-paste wrong issue number.

**Comment by jkiviluo** [comment]
> I'm not sure if the 'automatic environment matching' framework is feasible. Maybe a bit too ambitious.

Yes, this is definitely not needed at first. Should be separated to another issue and it can be done later. A linked idea is that whenever new item is linked to a specification, it could check if there is an applicable environment for that item and suggest to use that.

**Comment by jkiviluo** [comment]
This is mostly superseded by https://github.com/spine-tools/Spine-Toolbox/issues/3029, but keeping this open, since it has some pointers.
